6 router redux persist

nowherei opened the pull request:
Привет!

Настроил router. Подключил redux-persist. Проверь, пожалуйста.
Несколько моментов:
1) Мне сейчас не очень нравится как получились компоненты ChatList, MessageList. Может что-то подскажешь по улучшению или так тоже норм?
2) Наверно, хорошо было б смоделировать backend на тестовых данных. Тогда потом будет проще подключиться к api. Как это обычно делается? Или об этом можно пока не думать?

Как обычно, буду рад любой критике и совету по улучшению кода.

nowherei committed:
router, redux persist

nowherei committed:
Deleted constants from ChatList, MessageList.
Changed Header.

AlexandrV9 commented:
Всё отлично, но есть небольшие неточности:
1) При успешной регистрации, пользователя не перекидывает на страницу авторизации
2) У сообщений должен быть задний фон (сейчас они сливаются)
3) Лучше растянуть body и html на весь размер окна браузера: width: 100vw, height: 100vh
4) Почему тут **(index - 1)** -  у нас нет пользователей с userId равным -1, следовательно будет undefind
В компоненте MessageList:

        if (!index || array[index - 1]['userId'] !== userId) {
          data.image = userId === fromUserId ? fromUserImage : toUserImage;
        }
5) Про сервер поговорим на созвоне

AlexandrV9 commented:
Всё отлично, но есть небольшие неточности:
1) При успешной регистрации, пользователя не перекидывает на страницу авторизации
2) У сообщений должен быть задний фон (сейчас они сливаются)
3) Лучше растянуть body и html на весь размер окна браузера: width: 100vw, height: 100vh
4) Почему тут **index - 1:** - первый раз будет -1, у нас нет пользователей с userId равным -1, следовательно будет undefind
В компоненте MessageList:

        if (!index || array[index - 1]['userId'] !== userId) {
          data.image = userId === fromUserId ? fromUserImage : toUserImage;
        }
Я так понимаю, что это проверка отвечает за то, какая аватарка будет у пользователя. Проверка хорошая в данном случае, т.к у нас пока нету сервера
5) Про сервер обсудим на созвоне

AlexandrV9 reviewed (commented):
Данный этап проверен

nowherei commented:
> Всё отлично, но есть небольшие неточности:
> 
> 1. При успешной регистрации, пользователя не перекидывает на страницу авторизации
> 2. У сообщений должен быть задний фон (сейчас они сливаются)
> 3. Лучше растянуть body и html на весь размер окна браузера: width: 100vw, height: 100vh
> 4. Почему тут **index - 1:** - первый раз будет -1, у нас нет пользователей с userId равным -1, следовательно будет undefind
>    В компоненте MessageList:
>    ```
>     if (!index || array[index - 1]['userId'] !== userId) {
>       data.image = userId === fromUserId ? fromUserImage : toUserImage;
>     }
>    ```
> 
> Я так понимаю, что это проверка отвечает за то, какая аватарка будет у пользователя. Проверка хорошая в данном случае, т.к у нас пока нету сервера 5) Про сервер обсудим на созвоне

nowherei committed:
edited
1. Success registration redirect to authorization
2. MessageItem  added background
3. body, html - min-width: 100vw, min-height: 100vh

nowherei commented:
1-3. Поправил.

> 4. Почему тут index - 1: - первый раз будет -1, у нас нет пользователей с userId равным -1, следовательно будет undefind
> В компоненте MessageList:
```
 if (!index || array[index - 1]['userId'] !== userId) {
   data.image = userId === fromUserId ? fromUserImage : toUserImage;
 }
```
По-моему, здесь сейчас все верно работает. index - это index элемента в массиве. `!index || array[index - 1]['userId'] !== userId` - в этом условие идет проверка необходимости выводить аватарку. При index = 0, !index вернет true и до второй части проверки не дойдет. Пока ничего исправлять не стал.

5. По серверу пока ничего не делал.

Сделаю merge этой ветки в master.
